languages/tree-sitter-stack-graphs-typescript/test/jsx/jsx_core.tsx

```diff
@@ -0,0 +1,39 @@
+// The core of JSX tests here verify the behavior of the following node types:
+//   jsx_element
+//   jsx_identifier
+//   jsx_attribute
+//   jsx_expression
+//   jsx_opening_element
+//   jsx_closing_element
+// There is no real way to avoid testing all of these at once,
+// and so we don't even try to.
+
+let x = 1;
+
+// Flow In
+
+const el = <foo bar={x}>{x}</foo>;
+//                   ^ defined: 11
+//                       ^ defined: 11
+
+const el2 = <x></x>
```

Comment:
I'm not familiar with tsx, so this is a question for my understanding. The `x` here is a reference, but what about `foo` above? Is that an unresolved reference, or is it treated as a literal because it doesn't resolve to anything?

Reply:
In this example, `foo` is an undefined reference. The spec doesn't seem to define much beyond the grammar: https://facebook.github.io/jsx/.

For TSX, the TS handbook also defines the rules around type checking: https://www.typescriptlang.org/docs/handbook/jsx.html#type-checking
Specifically, the element tags need to be declared inside some interfaces and also there are rules regarding tags staring with lower case and upper case.